samples/apps/copilot-chat-app/webapi/Skills/ChatSkill.cs

```diff
@@ -534,26 +534,32 @@ private string OptimizeOpenApiSkillJson(string jsonContent, int tokenLimit, Plan
 
         List<object> itemList = new();
 
-        // Summary (List) Object
-        // To stay within token limits, attempt to truncate the list of results
-        if (document.RootElement.ValueKind == JsonValueKind.Array)
+        // Some APIs will return a JSON response with one property key representing an embedded answer
+        string resultsDescriptor = "";
+
+        if (document.RootElement.ValueKind == JsonValueKind.Object)
         {
-            foreach (JsonElement item in document.RootElement.EnumerateArray())
+            int propertyCount = 0;
+            foreach (JsonProperty property in document.RootElement.EnumerateObject())
             {
-                int itemTokenCount = Utilities.TokenCount(item.ToString());
+                propertyCount++;
+            }
 
-                if (jsonTokenLimit - itemTokenCount > 0)
-                {
-                    itemList.Add(item);
-                    jsonTokenLimit -= itemTokenCount;
-                }
-                else
-                {
-                    break;
-                }
+            if (propertyCount == 1)
+            {
+                // Save property name for result interpolation
+                JsonProperty firstProperty = document.RootElement.EnumerateObject().First();
+                jsonTokenLimit -= Utilities.TokenCount(firstProperty.Name);
+                resultsDescriptor = string.Format(CultureInfo.InvariantCulture, "{0}: ", firstProperty.Name);
+
+                // Extract object to be truncated
+                JsonElement value = firstProperty.Value;
+                document = JsonDocument.Parse(value.GetRawText());
             }
         }
 
+        // START Truncation
```

Comment:
Could you add more context to this comment (or remove it)?

Reply:
Removed it